### DISCUSSION TITLE
Optimize M2M Field

### DISCUSSION THREAD

**Discussion by jonasN5** [encode/django-rest-framework#9312] [General]
This is linked to https://github.com/encode/django-rest-framework/issues/4917

https://github.com/encode/django-rest-framework/blob/4c7c693f1555689cda88eb59a4353db69aa6f5b1/rest_framework/relations.py#L561

Why can't we just allow some customization at this line? Usually with any toolbox, we provide a best effort default and give the user the option to customize. In this case, the relations will ALWAYS be fetched with `.all()`. I have a simple case where I need an annotation on each child and so a single query is enough, if done on the `ManyRelatedField` level. How about making the `ManyRelatedField` also accept a `queryset` argument and default to `.all()` if there is no `queryset` argument?

**Comment by auvipy** [comment]
good point, can you share some PoC?

**Comment by jonasN5** [comment]
```
class ManyRelatedField(Field):
    ...
    queryset = None

    def __init__(self, child_relation=None, *args, **kwargs):
        self.queryset = kwargs.pop('queryset', self.queryset)
        ...

      

    def get_attribute(self, instance):
        # If a custom queryset has been set, use that before falling back to the best effort
        if self.queryset:
            return self.queryset
        ...
```

Super simple

Or explicit parameter
```
    def __init__(self, child_relation=None, queryset=None, *args, **kwargs):
        self.queryset = queryset
```

**Comment by jonasN5** [comment]
Any news on this? I can open a PR.
